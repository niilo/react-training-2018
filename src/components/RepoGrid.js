import React from 'react';
import PropTypes from 'prop-types';

const RepoGrid = props => {
  return (
    <ul className="popular-list">
      {props.repos.map((repo, index) => {
        return (
          <li key={repo.name} className="popular-item">
            <div className="popular-rank">#{index + 1}</div>
            <ul className="space-list-item">
              <li>
                <img
                  className="avatar"
                  src={repo.owner.avatar_url}
                  alt={`Avatar for ${repo.owner.login}`}
                />
              </li>
              <li>
                <a href={repo.html_url}>{repo.name}</a>
              </li>
              <li>@{repo.owner.login}</li>
              <li>{repo.stargazers_count} stars</li>
            </ul>
          </li>
        );
      })}
    </ul>
  );
};

RepoGrid.propTypes = {
  repos: PropTypes.array.isRequired,
};

export default RepoGrid;
