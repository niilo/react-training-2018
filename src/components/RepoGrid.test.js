import React from 'react';
import renderer from 'react-test-renderer';
import RepoGrid from './RepoGrid';

it('renders default correctly', () => {
  const props = {
    repos: [
      {
        name: 'react',
        html_url: 'https://github.com/facebook/react',
        stargazers_count: 73600,
        owner: {
          login: 'facebook',
          avatar_url: 'https://avatars3.githubusercontent.com/u/69631?v=4',
        },
      },
      {
        name: 'react2',
        html_url: 'https://github.com/facebook/react',
        stargazers_count: 73600,
        owner: {
          login: 'facebook',
          avatar_url: 'https://avatars3.githubusercontent.com/u/69631?v=4',
        },
      },
    ],
  };

  const tree = renderer.create(<RepoGrid {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
