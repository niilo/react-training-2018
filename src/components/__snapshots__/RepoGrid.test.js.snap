// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default correctly 1`] = `
<ul
  className="popular-list"
>
  <li
    className="popular-item"
  >
    <div
      className="popular-rank"
    >
      #
      1
    </div>
    <ul
      className="space-list-item"
    >
      <li>
        <img
          alt="Avatar for facebook"
          className="avatar"
          src="https://avatars3.githubusercontent.com/u/69631?v=4"
        />
      </li>
      <li>
        <a
          href="https://github.com/facebook/react"
        >
          react
        </a>
      </li>
      <li>
        @
        facebook
      </li>
      <li>
        73600
         stars
      </li>
    </ul>
  </li>
  <li
    className="popular-item"
  >
    <div
      className="popular-rank"
    >
      #
      2
    </div>
    <ul
      className="space-list-item"
    >
      <li>
        <img
          alt="Avatar for facebook"
          className="avatar"
          src="https://avatars3.githubusercontent.com/u/69631?v=4"
        />
      </li>
      <li>
        <a
          href="https://github.com/facebook/react"
        >
          react2
        </a>
      </li>
      <li>
        @
        facebook
      </li>
      <li>
        73600
         stars
      </li>
    </ul>
  </li>
</ul>
`;
